### PR TITLE
[E2E] Experiment running `downloads` test group as PR check

### PR DESCRIPTION
**Pull request by ariya**
Once this is merged, there is this check for every PR:

image

Note that the check will **only run** if the PR is already _marked for review_.

If the PR is **still** in draft, then the check list will look this (look at the `Skipped` indicator):

image

### CODE REVIEW

File: .github/workflows/e2e-tests.yml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    types: [opened, synchronize, reopened, ready_for_review]
+    paths-ignore:
+      - "docs/**"
+      - "**.md"
+      - ".circleci/**"
+
+jobs:
+
+  build:
+    if: github.event.pull_request.draft == false
```
Comment (diogormendes) [inline .github/workflows/e2e-tests.yml:14]: [❤️](https://emojipedia.org/red-heart/)

File: .github/workflows/e2e-tests.yml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    - run: ./bin/build frontend
+    - run: ./bin/build licenses
+    - run: ./bin/build drivers
+    - run: ./bin/build uberjar
+
+    - name: Prepare uberjar artifact
+      uses: ./.github/actions/prepare-uberjar-artifact
+
+  e2e-tests:
+    runs-on: buildjet-4vcpu-ubuntu-2004
```
Comment (diogormendes) [inline .github/workflows/e2e-tests.yml:41]: The` if: github.event.pull_request.draft == false` is only for the build Job, shouldn't we have it here as well ?

Comment (ariya) [reply .github/workflows/e2e-tests.yml:41]: No need, because of the `needs:` key.

**Review by diogormendes — APPROVED** [review]

File: .github/workflows/e2e-tests.yml (hunk 1 above)
Comment (nemanjaglumac) [reply .github/workflows/e2e-tests.yml:14]: Isn't this a duplicate of the

```yml
on:
  pull_request:
    types: [ready_for_review]
```

?

**Comment by nemanjaglumac** [comment]
I have just a few concerns:
1. How do we ensure we keep both `e2e` workflows in sync? (example: https://github.com/metabase/metabase/pull/21495)
2. Aren't we going to burn through our current `buildjet` plan REALLY fast once we start adding more groups (even though the trigger is set on the `pull_request`)?
    - We might start seeing queues again which would block people

File: .github/workflows/e2e-tests.yml (hunk 1 above)
Comment (nemanjaglumac) [reply .github/workflows/e2e-tests.yml:14]: Ooooooooh, no no no... I see what you did here :)

I was convinced that the default functionality for `ready_for_review` was basically the same as for `opened` IF the PR wasn't in the draft mode. It feels silly that `ready_for_review` doesn't trigger the workflow immediately after it's opened.

**Comment by diogormendes** [comment]
> I have just a few concerns:
> 
> 1. How do we ensure we keep both `e2e` workflows in sync? (example: [[E2E] Split Cypress run command on multiple lines #21495](https://github.com/metabase/metabase/pull/21495))
> 2. Aren't we going to burn through our current `buildjet` plan REALLY fast once we start adding more groups (even though the trigger is set on the `pull_request`)?
>    
>    * We might start seeing queues again which would block people

Agree with @nemanjaglumac on the queue risk. 
Since we are moving and stabilizing tests on GH Runners what do you think of us sticking with those?

File: .github/workflows/e2e-tests.yml (hunk 1 above)
Comment (ariya) [reply .github/workflows/e2e-tests.yml:14]: Yeah, that was my first attempt and it didn't work well. Hence, the circus dance with various `pull_request` triggers.

**Comment by ariya** [comment]
> 1. How do we ensure we keep both `e2e` workflows in sync? (example: [[E2E] Split Cypress run command on multiple lines #21495](https://github.com/metabase/metabase/pull/21495))

They are different anyway, this one only uses `--folder` (very simple invocation).
At some point once we merge the two workflows, I believe the duplication is gone (they have to converge).

> 2. Aren't we going to burn through our current `buildjet` plan REALLY fast once we start adding more groups (even though the trigger is set on the `pull_request`)?
>    * We might start seeing queues again which would block people

Isn't the purpose of the experiment to find that out? Otherwise, we can only run pull requests checks using hosted runners (slow, but manageable) or if the upper limit of Buildjet resources is already increased.

Technically, PR won't be blocked since none of the E2E tests are `Required` checks. But yeah, people might be still waiting for the run to finish regardless.

> Edit: Also, aren't we supposed to remove `downloads` from the CircleCI now?

My original thought was removing CircleCI after the test group is known to be working with GitHub Actions first (either one by one, or in a group). But I'm fine if we are doing immediate _moving_ (for each test group) instead of _duplicating-and-deleting_ (as two separate steps).

**Review by nemanjaglumac — APPROVED** [review]
Let's do this! 🚀 

@ariya completely up to you if you want to remove `downloads` from CircleCI now or if you want to wait.

**Pull request merged**